docs/esp32/esp32-state.md

```diff
@@ -0,0 +1,141 @@
+# ESP32 State
+
+In this tutorial, you can monitor WiFi status (SSID and RSSI) surrounding the ESP32 device and let the device restart using Azure IoT Hub device twins.
+
+## What you need
+
+Finish the [Getting Started Guide](./esp32-get-started.md) to:
+
+- Get basic knowledge of ESP32 device.
+- Prepare the development environment.
+
+
+An active Azure subscription. If you do not have one, you can register via one of these two methods:
+
+- Activate a [free 30-day trial Microsoft Azure account](https://azure.microsoft.com/free/).
+- Claim your [Azure credit](https://azure.microsoft.com/pricing/member-offers/msdn-benefits-details/) if you are MSDN or Visual Studio subscriber.
+
+
+## Open the project folder
+
+### Start VS Code
+
+- Start Visual Studio Code.
+- Make sure [Azure IoT Workbench](https://marketplace.visualstudio.com/items?itemName=vsciot-vscode.vscode-iot-workbench) is installed.
+- Connect ESP32 to your PC.
+
+### Open IoT Workbench Examples
+
+Use `F1` or `Ctrl+Shift+P` (macOS: `Cmd+Shift+P`) to open the command palette, type **IoT Workbench**, and then select **IoT Workbench: Examples**.
+
+![IoT Workbench: Examples](media/iot-workbench-examples-cmd.png)
+
+Select **ESP32 Arduino**.
+
+![IoT Workbench: Examples -> Select board](media/iot-workbench-examples-board.png)
+
+Then the **IoT Workbench Example** window is shown up.
+
+![IoT Workbench, Examples window](media/iot-workbench-examples.png)
+
+Find **ESP32 State** and click **Open Sample** button. A new VS Code window with a project folder in it opens.
+
+![IoT Workbench, select ESP32 State example](media/esp32-state/open-example-esp32state.jpg)
```

Comment:
Is this pic correct?

Reply:
No. Tanya is working on the example images. I will update the pictures once I get those images.